### PR TITLE
added did:key provider; fixes #335

### LINKED ISSUE #335: DIDManager: Implementation for did:key
Add IdentifierProvider for did:key

### PR DESCRIPTION
**Pull request by awoie**
added did:key provider; fixes #335

### CODE REVIEW

File: packages/did-provider-key/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Change Log
```
Comment (mirceanis) [inline packages/did-provider-key/CHANGELOG.md:1]: This file should be removed

File: packages/did-provider-key/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Provides `did:web` {@link @veramo/did-provider-key#KeyDIDProvider | identifier provider } for the
```
Comment (mirceanis) [inline packages/did-provider-key/src/index.ts:2]: This should reference did:key, not did:web
```suggestion
 * Provides `did:key` {@link @veramo/did-provider-key#KeyDIDProvider | identifier provider } for the
```

**Review by mirceanis — CHANGES_REQUESTED** [review]
Nice work!

I flagged some tiny changes that need to be done and it would also be nice to see some tests with this, like checking that a credential issued by a `did:key` looks ok, or that calling the CRUD methods to modify the DID document throw expected errors.

**Comment by jasonhealy** [comment]
@awoie If you want the provider to be documented you  need to add it in here: https://github.com/uport-project/veramo/blob/awoie/did-key/docsconfig.json

example doc: http://veramo.io/docs/api/did-provider-ethr.ethrdidprovider

**Review by mirceanis — APPROVED** [review]
looks good

**Pull request merged**